full-stack-practice/server/server.ts

```diff
@@ -0,0 +1,61 @@
+import 'dotenv/config';
+import pg from 'pg';
+import express from 'express';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+type Todo = {
+  entryId: number;
+  task: string;
+  isCompleted: boolean;
+  createdAt: string;
+  updatedAt: string;
+};
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/products', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "products"
+        order by "productId";
+    `;
+    const result = await db.query<Todo>(sql);
+    res.json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/products/:productId', async (req, res, next) => {
+  try {
+    const { productId } = req.params;
```

Comment:
You should check that `productId` is an integer. Although your client will always send you an integer, someone might be hacking you and call your endpoint with an invalid productId.